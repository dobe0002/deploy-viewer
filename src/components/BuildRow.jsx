import React from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'reactstrap';
import Status from './Status';

const BuildRow = props => {
  const { appTitle, appStatus } = props;

  const rowStyle = {
    borderTop: '1px solid #ccc',
    paddingTop: '15px',
    paddingBottom: '15px'
  };
  const iconStyle = { marginRight: '10px', display: 'inline-block' };

  return (
    <Row style={rowStyle}>
      <Col xs='2'>
        <h2 className='h5'>{appTitle}</h2>
      </Col>
      <Col>
        {appStatus.map(status => {
          if (props.startDate !== '' && status.date < props.startDate) {
            return '';
          }
          if (props.endDate !== '' && status.date > props.endDate) {
            return '';
          }
          return (
            <div style={iconStyle} key={status.row}>
              <Status
                id={`popper${status.row}`}
                code={status.status}
                description={status.statusDescription}
                appTitle={appTitle}
                icon
                popOver
                dateString={status.dateString}
                note={status.note}
              />
            </div>
          );
        })}
      </Col>
    </Row>
  );
};

BuildRow.propTypes = {
  appTitle: PropTypes.string.isRequired,
  appStatus: PropTypes.arrayOf.isRequired,
  startDate: PropTypes.instanceOf(Date),
  endDate: PropTypes.instanceOf(Date)
};
BuildRow.defaultProps = {
  startDate: '',
  endDate: ''
};

export default BuildRow;
