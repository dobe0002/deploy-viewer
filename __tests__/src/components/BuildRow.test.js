/* eslint-disable react/jsx-filename-extension */
import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import BuildRow from '../../../src/components/BuildRow';

configure({ adapter: new Adapter() });

const { axe, toHaveNoViolations } = require('jest-axe');

expect.extend(toHaveNoViolations);

describe('Build Rows tests', () => {
  const sampleProps = {
    appTitle: 'My application',
    appStatus: [
      {
        row: 3,
        status: 'G',
        statusDescription: 'Good deployment',
        dateString: '11/02/2019',
        date: new Date(2019, 10, 2),
        note: 'Hooray everything went to plan'
      },
      {
        row: 5,
        status: 'O',
        statusDescription: 'Outage',
        dateString: '11/10/2019',
        date: new Date(2019, 10, 10),
        note: 'People are trashing the database'
      }
    ]
  };
  test('Accessibility check', async () => {
    const wrapper = shallow(
      <main>
        <BuildRow
          appTitle={sampleProps.appTitle}
          appStatus={sampleProps.appStatus}
        />
      </main>
    ); // Main tag is needed to prevent landmark errors
    const results = await axe(wrapper.html());
    // console.log('Axe violations', results.violations);
    expect(results).toHaveNoViolations();
    // console.log(wrapper.html());
  });
  test('Change start date', () => {
    const wrapper = shallow(
      <BuildRow
        appTitle={sampleProps.appTitle}
        appStatus={sampleProps.appStatus}
        startDate={new Date(2019, 10, 3)}
      />
    );
    expect(wrapper.find('Status')).toHaveLength(1);
    expect(wrapper.find('Status').prop('code')).toEqual('O');
  });
  test('Change end date', () => {
    const wrapper = shallow(
      <BuildRow
        appTitle={sampleProps.appTitle}
        appStatus={sampleProps.appStatus}
        endDate={new Date(2019, 10, 3)}
      />
    );
    expect(wrapper.find('Status')).toHaveLength(1);
    expect(wrapper.find('Status').prop('code')).toEqual('G');
  });

  test('Snapshot test', () => {
    const wrapper = shallow(
      <BuildRow
        appTitle={sampleProps.appTitle}
        appStatus={sampleProps.appStatus}
      />
    );
    expect(wrapper.find('Status')).toHaveLength(2);
    expect(wrapper.find('h2').text()).toEqual(sampleProps.appTitle);
    expect(wrapper).toMatchSnapshot();
  });
});
