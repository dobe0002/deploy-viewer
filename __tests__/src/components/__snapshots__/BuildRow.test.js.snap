// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Build Rows tests Snapshot test 1`] = `
<Row
  style={
    Object {
      "borderTop": "1px solid #ccc",
      "paddingBottom": "15px",
      "paddingTop": "15px",
    }
  }
  tag="div"
  widths={
    Array [
      "xs",
      "sm",
      "md",
      "lg",
      "xl",
    ]
  }
>
  <Col
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
    xs="2"
  >
    <h2
      className="h5"
    >
      My application
    </h2>
  </Col>
  <Col
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <div
      key="3"
      style={
        Object {
          "display": "inline-block",
          "marginRight": "10px",
        }
      }
    >
      <Status
        appTitle="My application"
        code="G"
        dateString="11/02/2019"
        description="Good deployment"
        icon={true}
        id="popper3"
        note="Hooray everything went to plan"
        popOver={true}
        statuses={Array []}
        tableCell={false}
      />
    </div>
    <div
      key="5"
      style={
        Object {
          "display": "inline-block",
          "marginRight": "10px",
        }
      }
    >
      <Status
        appTitle="My application"
        code="O"
        dateString="11/10/2019"
        description="Outage"
        icon={true}
        id="popper5"
        note="People are trashing the database"
        popOver={true}
        statuses={Array []}
        tableCell={false}
      />
    </div>
  </Col>
</Row>
`;
